### PR TITLE
Add convenience methods for HomStructure

### PR DESCRIPTION
**Pull request by sebastianpos**

### CODE REVIEW

File: CAP/gap/CategoryMorphisms.gd [hunk 1]
```diff
@@ -1952,6 +1952,43 @@ DeclareOperation( "AddMereExistenceOfSolutionOfLinearSystemInAbCategory",
...
+                  [ IsCapCategoryMorphism, IsCapCategoryMorphism] );
+
+#! @Description
+#! This is a convenience method.
+#! The arguments are a morphism $\alpha: a \rightarrow a'$ and an object $b$ in $C$.
+#! The output is <C>HomomorphismStructureOnMorphisms</C> called on $\alpha$, $\mathrm{id}_b$.
+#! @Returns a morphism in $\mathrm{Hom}_{D}(H(a',b), H(a,b))$
+#! @Arguments alpha, b
+DeclareOperation( "HomStructure",
+                  [ IsCapCategoryObject, IsCapCategoryMorphism] );
```
Comment (mohamed-barakat) [inline CAP/gap/CategoryMorphisms.gd]: Please interchange the arguments to comply with the documentation.
```suggestion
                  [ IsCapCategoryMorphism, IsCapCategoryObject ] );
```

File: CAP/gap/CategoryMorphisms.gd [hunk 2]
```diff
@@ -1952,6 +1952,43 @@ DeclareOperation( "AddMereExistenceOfSolutionOfLinearSystemInAbCategory",
...
+
+#! @Description
+#! This is a convenience method.
+#! The arguments are an object $a$ and a morphism $\beta: b \rightarrow b'$ in $C$.
+#! The output is <C>HomomorphismStructureOnMorphisms</C> called on $\mathrm{id}_a$, $\beta$.
+#! @Returns a morphism in $\mathrm{Hom}_{D}(H(a,b), H(a,b'))$
+#! @Arguments a, beta
+DeclareOperation( "HomStructure",
+                  [ IsCapCategoryMorphism, IsCapCategoryObject] );
```
Comment (mohamed-barakat) [inline CAP/gap/CategoryMorphisms.gd]: Please interchange the arguments to comply with the documentation.
```suggestion
                  [ IsCapCategoryObject, IsCapCategoryMorphism ] );
```

File: CAP/gap/CategoryMorphisms.gd [hunk 3]
```diff
@@ -1952,6 +1952,43 @@ DeclareOperation( "AddMereExistenceOfSolutionOfLinearSystemInAbCategory",
...
                   [ IsCapCategory, IsList ] );
 
+#! @Description
+#! This is a convenience method.
+#! The arguments are two morphisms $\alpha: a \rightarrow a', \beta: b \rightarrow b'$ in $C$.
+#! The output is <C>HomomorphismStructureOnMorphisms</C> called on $\alpha$, $\beta$.
+#! @Returns a morphism in $\mathrm{Hom}_{D}(H(a',b), H(a,b'))$
+#! @Arguments alpha, beta
+DeclareOperation( "HomStructure",
+                  [ IsCapCategoryMorphism, IsCapCategoryMorphism] );
```
Comment (mohamed-barakat) [inline CAP/gap/CategoryMorphisms.gd]: Whitespace change:
```suggestion
                  [ IsCapCategoryMorphism, IsCapCategoryMorphism ] );
```

File: CAP/gap/CategoryMorphisms.gd [hunk 4]
```diff
@@ -1952,6 +1952,43 @@ DeclareOperation( "AddMereExistenceOfSolutionOfLinearSystemInAbCategory",
...
+                  [ IsCapCategoryMorphism, IsCapCategoryObject] );
+
+#! @Description
+#! This is a convenience method.
+#! The arguments are two objects $a$ and $b$ in $C$.
+#! The output is <C>HomomorphismStructureOnObjects</C> called on $a,b$.
+#! @Returns an object
+#! @Arguments a, b
+DeclareOperation( "HomStructure",
+                  [ IsCapCategoryObject, IsCapCategoryObject] );
```
Comment (sebastianpos) [inline CAP/gap/CategoryMorphisms.gd]:
```suggestion
                  [ IsCapCategoryObject, IsCapCategoryObject ] );
```

**Comment by sebastianpos** [comment]
Thanks for spotting these typos.

**Pull request merged**